src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Server/ServerPrototype.cs

```diff
@@ -425,6 +453,35 @@ public bool UpdateMemoryValues(Microsoft.SqlServer.Management.Smo.Server server)
             return changesMade;
         }
 
+        public bool UpdateSecurityValues(Server server)
+        {
+            bool alterServer = false;
+            bool needServerRestart = false;
+
+            if (this.currentState.AuthenticationMode != this.originalState.AuthenticationMode)
+            {
+                // set authentication
+                server.Settings.LoginMode = this.currentState.AuthenticationMode;
+                alterServer = true;
+                needServerRestart = true;
+            }
+
+            if (this.currentState.LoginAuditing != this.originalState.LoginAuditing)
+            {
+                server.Settings.AuditLevel = this.currentState.LoginAuditing;
+                alterServer = true;
+            }
+
+            if (needServerRestart)
+            {
+                // user is applying the changes right now (not scripting/scheduling)
+                // and changes require server restart in order to be effective so
+                // we warn the user about the required restart action (see also SQLBU# 355718)
+                Logger.Information("Changes require server restart in order to be effective");
```

Comment:
Does this need to be localized?

Reply:
@barbaravaldez we should be doing something on the ADS side so it shows a toast notification. Can you move this check over there instead?

Reply:
Or maybe it would be better as a message in the dialog...should chat with @erinstellato-ms and come up with a design and we can review it.

Reply:
I voted for message (toast notifications are easily missed by folks not familiar with ADS/VS Code)